php/db.php

```diff
@@ -73,6 +76,103 @@ function GetDataForTable($tabName){
 	return $tabData;
 }
 
+function MigrateDatabase() {
+	global $dbConn, $ip, $userAgent, $dbVersion;
+	$migrationsDir = "SQL/versions/";
+
+
+	StartTimer("MigrateDatabase");
+
+	// Check our current database version
+	$sql = "SELECT config_value FROM config WHERE config_key = 'DATABASE_VERSION';";
+	$data = mysqli_query($dbConn, $sql);
+	
+	$config_result = mysqli_fetch_array($data);
+
+	if ($config_result == NULL) {
+		die("Unable to determine the database version. Please add the config value DATABASE_VERSION before running");
+	}
+
+	$currentDatabaseVersion = intval($config_result[0], 10);
+
+	// Check and see if we need to migrate.
+	if ($dbVersion < $currentDatabaseVersion) {
+		die('Database version is out of date with code version');
+	} elseif ($dbVersion == $currentDatabaseVersion) {
+		StopTimer("MigrateDatabase");
+		return;
+	}
+	
+	// Calculate what migrations exist
+	$migrations = array();
+	
+	$migrationDirFiles = scandir($migrationsDir);
+	foreach ($migrationDirFiles as $i => $migrationFile) {
+		// Get the version ID from the filename. Files need to be formatted <versionID>_<randomName>.sql
+		preg_match('/(\d+)_.*\.sql$/', $migrationFile, $matches);
+		if (isset($matches[1])) {
+			$migrationID = intval($matches[1], 10);
```

Comment:
If I understand this right, the database version is then defined by the name of the migration file?

Reply:
The migration version is defined by the file name. The database version is defined by dbVersion at the top of the file.

The script won't run any migration with an ID higher (or lower) than the target database version.